### DISCUSSION TITLE
why not add scaffoldings for users to register, login, reset password etc.

### DISCUSSION THREAD

**Discussion by ghosttk** [pradosoft/prado#863] [Polls]
Just like Breeze in Laravel or Shield in Codeigniter.

**Comment by belisoful** [comment]
This is the work of a proper TWebDBUserManager.  I've actually been looking at implementing this for a while.  as a composer extension.

I have hefty work on The new asset publishing system, cron v2, system updates, and email module.

In my view, these are higher priority than a user manager....  but yes.  registration, login, reset pwd, etc are part of the system that needs to be "upgraded" for 2023.

I'll point out that the TWebUserManager, like you suggest, relies upon sending emails... which there is no dedicated subsystem in PRADO yet.  That is what I am working and your point is actually why.

**Comment by belisoful** [comment]
I actually have a REALLY well coded User Manager module to do some of what you suggest.  Again, the lack of email support in PRADO is an issue.  Email in PRADO I am actively worked on.

**Comment by belisoful** [comment]
It's also not possible with standardized Web pages.   Which can be done now with composer PRADO plugins.  Plugins can be their own module and have their own pages.  When email is done, they can have their own emails too.

standard PRADO doesn't have its own pages.  At least not yet.  Composer plugins are the answer to that.  Meaning, I think your suggestion is GREAT as a PRADO composer idea.  So much so,  I have plans on doing it.

Also, a great TWebUserModule wouldn't be possible without behaviors, which was added in 3.2.2 and is still a rarely used feature.

**Reply by ghosttk** [reply]
Thank you for your hard work. 
I'm new to PRADO framework and love it very much.
I'm really looking forward to its new features.
